src/pool/srv_pool.c

```diff
@@ -1458,6 +1458,47 @@ resume_event_handling(struct pool_svc *svc)
 	ABT_mutex_unlock(events->pse_mutex);
 }
 
+/*
+ * Restart rebuild if the rank is UPIN in pool map and is in rebuilding.
+ *
+ * This function only used when PS leader gets CRT_EVT_ALIVE event of engine \a rank,
+ * if that rank is UPIN in pool map and with unfinished rebuilding should be massive
+ * failure case -
+ * 1. some engines down and triggered rebuild.
+ * 2. the engine \a rank participated the rebuild, not finished yet, it became down again,
+ *    the #failures exceeds pool RF and will not change pool map.
+ * 3. That engine restarted by administrator.
+ *
+ * In that case should recover the rebuild task on engine \a rank, to simplify it now just
+ * abort and retry the global rebuild task.
+ */
+static void
+pool_restart_rebuild_if_rank_wip(struct ds_pool *pool, d_rank_t rank)
+{
+	struct pool_domain	*dom;
+
+	ABT_rwlock_rdlock(pool->sp_lock);
+	dom = pool_map_find_dom_by_rank(pool->sp_map, rank);
+	if (dom == NULL) {
+		ABT_rwlock_unlock(pool->sp_lock);
+		D_INFO(DF_UUID": rank %d non-exist on pool map.\n",
+		       DP_UUID(pool->sp_uuid), rank);
```

Comment:
[Nit] If there are pools who do not comprise _all_ engines in a system, we may get this `INFO` often, and it is not very useful. So I think this should be a `D_DEBUG`.